dev/Dockerfile.test

```diff
@@ -6,22 +6,33 @@ ENV DEBIAN_FRONTEND=noninteractive \
 
 WORKDIR /app
 
-ADD . /app
-
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends build-essential openjdk-11-jre-headless \
+    apt-get install -y --no-install-recommends build-essential openjdk-17-jre-headless \
     # cmake and protobuf-compiler required for onnx install
     cmake protobuf-compiler && \
     # Install miniforge
     wget -q https://github.com/conda-forge/miniforge/releases/latest/download/$MINIFORG_FILENAME && \
     bash $MINIFORG_FILENAME -b -p /opt/conda && \
-    rm $MINIFORG_FILENAME && \
+    rm $MINIFORG_FILENAME
+
+ADD ./requirements /app/requirements
+
+RUN \
     # install required python packages
-    pip install --no-cache-dir -r requirements/test-requirements.txt --no-cache-dir -r requirements/lint-requirements.txt && \
+    pip install -r requirements/test-requirements.txt --no-cache-dir \
+        -r requirements/lint-requirements.txt psutil
```

Comment:
Fix: `tests/server/auth/test_auth.py` requires `psutil`. I had the same issue with `dev/dev-env-setup.sh`, but there I simply installed the package manually. I tried adding `psutil` to `requirements/test-requirements.txt`, but this test failed after working in an earlier PR build (may or may not have been caused by my change):
```
tests/evaluate/test_default_evaluator.py::test_evaluation_with_env_restoration[conda]
```